### ISSUE TITLE
How to resume training from batch index

### ISSUE THREAD

**Issue by anjul1008** [wenet-e2e/wenet#2565]
Hi, 
I saw new step the checkpoint saving feature recently. 
I noticed training starts from batch index 0, how to resume training from the same batch index where i stopped the training?

I didn't find the direct way as it happens in pytorch dataloader.
Can you provide some suggestion?

thank you.

**Comment by xingchensong** [comment]
its not supported yet. BTW，resume from batch index0 dose not hurt performance in our experiments

**Comment by xingchensong** [comment]
a quick solution is shuffling data.list with different seed before every resume

**Comment by anjul1008** [comment]
Yeah, but we can't be sure that we iterated over the entire data in each epoch.
I'm thinking of manipulating epoch parameter and iterating over small subsets as an epoch.(epoch = compete_data_epoch * #subsets)

but it's just a hack.

**Comment by xingchensong** [comment]
why do need to stop several times in a week

**Comment by anjul1008** [comment]
Due to resource limitations.

I have observed that the training process terminates after running a few batches because one of the GPUs reaches out-of-memory (OOM) while the other GPUs still have available memory. This indicates that the batching process is not utilizing the full potential of all GPUs efficiently.

I have tried all 3 batching strategies, but the behavior remains same.

**Comment by xingchensong** [comment]
1. decrease batchsize and increase accum_grad
2. use deepspeed/fsdp
3. turn on gradient checkpointing
